ScratchWikiSkinTemplate.php

```diff
@@ -103,6 +103,37 @@ public function execute() {
 					</div>
 				</form>
 			</li>
+			<?php
+			//  Start dropdown here so we know its output before it saves to the page, and we can use it earlier.
+			$swsDropdownItems = "";
+			foreach ($this->data['personal_urls'] as $key => $tab) { 
+					if (!array_key_exists('href', $tab)) {
+						if (!array_key_exists('class', $tab)) {
+							$tab['class'] = '';
+						}
+						
+						$tab['class'] .= 'no-link';
+					}
+					if(isset($tab["class"]) && gettype($tab["class"]) == "array" && $tab["class"][0] == "mw-echo-alert"){
+						$swsUnread = true;
+						continue;
+					}
+					if(strpos($key, "notifications") !== false){
+						if(in_array("mw-echo-unseen-notifications", $tab["link-class"])){
+							$swsUnread = true;
+						}
+						continue;
+					}
+					$swsDropdownItems .= $this->getSkin()->makeListItem($key, $tab);
+			}
+			// ONLY output this if Echo extension is turned on. Otherwise don't put it there.
+			if(ExtensionRegistry::getInstance()->isLoaded("Echo")){
+			?>
+			<li class="link right notifications">
+				<a href="/wiki/Special:Notifications<?php /* This probably should be localized and not hardcoded, but it works*/ ?>"><div></div></a>
```

Comment:
You can (and should) use [`$this->makeLink`](https://doc.wikimedia.org/mediawiki-core/master/php/classBaseTemplate.html#a571b83eb61a5ed0bb91522c2a33eabd7)

Reply:
Not sure if that works here because it's an arbitrary page and there's no data. But I was able to localize the page using the same way the other links are